src/Simulation/CollisionModule.lua

```diff
@@ -347,13 +348,10 @@ function module:FetchHullsForBox(min, max)
 	
 	--Inflate missing hulls
 	for key,record in pairs(hullRecords) do
-       
     	if (record.hull == nil) then
-			record.hull = self:GenerateConvexHullAccurate(record.instance, module.expansionSize, self:GenerateSnappedCFrame(record.instance))
-            if (record.hull == nil) then
-                hullRecords[key] = nil
-            end
-		end
+            record.hull = {}
+        end
+		record.hull[fetchExpansionSize] = self:GenerateConvexHullAccurate(record.instance, fetchExpansionSize, self:GenerateSnappedCFrame(record.instance))
```

Comment:
You shouldn't regenerate this every time. Should be a if record.hull[fetchExpansionSize] == nil check here!

Reply:
Ah thank you, good catch!